src/views/studio/CreateMemberView/CreateMemberView.tsx

```diff
@@ -133,10 +138,25 @@ const CreateMemberView = () => {
               required: true,
               pattern: MEMBERSHIP_NAME_PATTERN,
               patternMessage: 'may contain only lowercase letters, numbers and underscores',
+              validate: async (value) => {
+                // Wrapping it up with promise and resolving comparison inside debounce
+                // debounce() will not automatically do the return, which is needed for validation
+                return new Promise((resolve) => {
+                  debounce(async (handle) => {
+                    const {
+                      data: { membershipByUniqueInput },
+                    } = await refetchMember()
+                    resolve(membershipByUniqueInput?.handle !== handle)
+                  }, 500)(value)
+                })
+              },
             })
           )}
+          onChange={async () => await trigger('handle')}
```

Comment:
Do we need that line? I think `react-hook-form` will automatically trigger validation on submit? The side effect of this is that user will get error messages when typing (for example "handle too short" when I just started typing). It's a bad UX to display errors too early. If they submit once, the error checking mode should change to `onChange` and the user should get updated right away

Reply:
Good point, removed that `onChange`